pkg/acsengine/defaults.go

```diff
@@ -584,8 +591,25 @@ func setMasterProfileDefaults(a *api.Properties, isUpgrade bool) {
 	}
 }
 
-// setVMSSDefaults
-func setVMSSDefaults(a *api.Properties) {
+// setVMSSDefaultsForMasters
+func setVMSSDefaultsForMasters(a *api.Properties) {
+	if a.MasterProfile.Count > 100 {
+		a.MasterProfile.SinglePlacementGroup = helpers.PointerToBool(false)
+	}
+	if a.MasterProfile.SinglePlacementGroup == nil {
+		a.MasterProfile.SinglePlacementGroup = helpers.PointerToBool(api.DefaultSinglePlacementGroup)
+	}
+	if a.MasterProfile.SinglePlacementGroup == helpers.PointerToBool(false) {
+		a.MasterProfile.StorageProfile = api.ManagedDisks
+	}
+	if a.MasterProfile.HasAvailabilityZones() {
```

Comment:
same here for overriding LB defaults

Reply:
Same here.

Reply:
now it's a validation instead of default